File: src/main/java/com/alealogic/singleproxy/service/RegistrationService.java
package com.alealogic.singleproxy.service;

import com.alealogic.singleproxy.repository.DesktopClientRepository;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

@Slf4j
@Service
public class RegistrationService {

    private final DesktopClientRepository desktopClientRepository;

    public RegistrationService(DesktopClientRepository desktopClientRepository) {
        this.desktopClientRepository = desktopClientRepository;
    }

    public void registerDesktopClient(String desktopClientKey) {
        final var byKey = desktopClientRepository.findByKey(desktopClientKey);
        final var desktopClient = byKey
                .orElseThrow(() -> {
                    log.info("no DesktopClient found in db with desktopClientKey " + desktopClientKey);
                    return new IllegalStateException("no ");
                });

        desktopClient.setRegistered(true);
        desktopClientRepository.save(desktopClient);
    }
}

File: src/main/java/com/alealogic/singleproxy/service/RegistrationService.kt
package com.alealogic.singleproxy.service

import com.alealogic.singleproxy.repository.DesktopClientRepository
import org.springframework.stereotype.Service

@Service
class RegistrationService(private val desktopClientRepository: DesktopClientRepository) {

    fun registerDesktopClient(desktopClientKey: String) {
        val desktopClient = desktopClientRepository.findByKey(desktopClientKey)
        if (desktopClient != null) {
            desktopClient
                .apply { registered = true }
                .also { desktopClientRepository.save(it) }
        } else throw IllegalStateException("no DesktopClient found in db with desktopClientKey $desktopClientKey")

        desktopClient.updateInitiatorDesktopClientId?.let {
            desktopClientRepository.findByUpdateInitiatorDesktopClientId(it)
        }?.apply { obsolete = true }?.also { desktopClientRepository.save(it) }
    }
}
